src/server/mod.rs

```diff
@@ -386,6 +386,7 @@ impl<O: Output> LsService<O> {
         while self.handle_message() == ServerStateChange::Continue {}
     }
 
+    #[allow(cyclomatic_complexity)] // complexity after macro expansion
```

Comment:
can we just globally disable this lint?

Reply:
Up to you if you don't think it's useful. Here it's triggered by the complexity _after_ macro expansion, which actually isn't really important and that's why I allow it here explicitly. But in general I think the rule works.

Reply:
I think if a function is too complex, it gets caught in review, we don't need Clippy to check it.

Reply:
ok